lib/BackbeatMetadataProxy.js

```diff
@@ -40,27 +40,30 @@ class BackbeatMetadataProxy extends BackbeatTask {
         return new RoleCredentials(vaultclient, extension, role, log);
     }
 
-    putMetadata(entry, log, cb) {
+    putMetadata(params, mdBlob, log, cb) {
```

Comment:
Seems we could make the interface a bit simpler by making `mdBlob` one of the properties of `params`. But that could end up making the logging a bit more complex. Just thinking out loud about trade-offs.

Reply:
I agree, it could make the API slightly simpler, for logging it's a minor concern and passing the fields to log explicitly is fine as there are only three of them.